jinja2/runtime.py

```diff
@@ -422,29 +422,17 @@ class LoopContext(LoopContextBase):
 
     def __init__(self, iterable, undefined, recurse=None, depth0=0):
         LoopContextBase.__init__(self, undefined, recurse, depth0)
-        self._iterator = iter(iterable)
-
-        # try to get the length of the iterable early.  This must be done
-        # here because there are some broken iterators around where there
-        # __len__ is the number of iterations left (i'm looking at your
-        # listreverseiterator!).
         try:
+            # Getting length of iterable
+            iterable = tuple(iterable)
             self._length = len(iterable)
+            self._iterator = iter(iterable)
         except (TypeError, AttributeError):
             self._length = None
         self._after = self._safe_next()
 
     @property
     def length(self):
```

Comment:
Maybe this one can be moved to the base class? Since there in the parent class dependencies of this property?